domain/src/test/java/org/oppia/domain/oppialogger/analytics/AnalyticsControllerTest.kt

```diff
@@ -358,13 +399,108 @@ class AnalyticsControllerTest {
     assertThat(eventContext.revisionCardContext.subTopicId).matches(TEST_SUB_TOPIC_ID)
   }
 
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logTransitionEvent_withNoNetwork_checkLogsEventToStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    analyticsController.logTransitionEvent(
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createQuestionContext(
+        TEST_QUESTION_ID,
+        listOf(
+          TEST_SKILL_LIST_ID, TEST_SKILL_LIST_ID
+        )
+      )
+    )
+
+    val eventLogs = analyticsController.getEventLogs()
+    eventLogs.observeForever(this.mockOppiaEventLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(
+      this.mockOppiaEventLogsObserver,
+      atLeastOnce()
+    ).onChanged(oppiaEventLogsResultCaptor.capture())
+
+    val eventLog = oppiaEventLogsResultCaptor.value.getOrThrow().getEventLog(0)
+    assertThat(eventLog.priority).isEqualTo(Priority.ESSENTIAL)
+    assertThat(eventLog.context.activityContextCase).isEqualTo(QUESTION_CONTEXT)
+    assertThat(eventLog.timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(eventLog.actionName).isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logClickEvent_withNoNetwork_checkLogsEventToStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    analyticsController.logClickEvent(
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createQuestionContext(
+        TEST_QUESTION_ID,
+        listOf(
+          TEST_SKILL_LIST_ID, TEST_SKILL_LIST_ID
+        )
+      )
+    )
+
+    val eventLogs = analyticsController.getEventLogs()
+    eventLogs.observeForever(this.mockOppiaEventLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+    verify(
+      this.mockOppiaEventLogsObserver,
+      atLeastOnce()
+    ).onChanged(oppiaEventLogsResultCaptor.capture())
+
+    val eventLog = oppiaEventLogsResultCaptor.value.getOrThrow().getEventLog(0)
+    assertThat(eventLog.priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(eventLog.context.activityContextCase).isEqualTo(QUESTION_CONTEXT)
+    assertThat(eventLog.timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(eventLog.actionName).isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logTransitionEvent_withNoNetwork_exceedLimit_checkEventLogStoreSize() {
```

Comment:
Other tests that might be useful to verify:
- Uploading the events clears the queue
- Exceeding the clear, then uploading some, allows more events to be logged
- Both click and transition events can be logged at the same time
- The correct old event is evicted from the queue
- The logs are uploaded in the correct order

Reply:
Actually, the functionality for uploading events from the store to the remote service is a part of M3. So I will add tests related to uploading of events then. Does that make sense ?

Reply:
Added 3 test cases to increase test thoroughness as per the comment.

Reply:
Just to clarify that tests 1 & 2 mentioned above will be part of M3 (@Sarthak2601 can you add a TODO with a bug here?) Verified that the other 3 are tested. Resolving this